src/main/java/de/tum/cit/aet/artemis/coursenotification/web/UserCourseNotificationStatusResource.java

```diff
@@ -0,0 +1,52 @@
+package de.tum.cit.aet.artemis.coursenotification.web;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.cit.aet.artemis.core.repository.UserRepository;
+import de.tum.cit.aet.artemis.coursenotification.dto.UserCourseNotificationStatusUpdateRequestDTO;
+import de.tum.cit.aet.artemis.coursenotification.service.UserCourseNotificationStatusService;
+
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/coursenotification/")
+public class UserCourseNotificationStatusResource {
+
+    private static final Logger log = LoggerFactory.getLogger(UserCourseNotificationStatusResource.class);
+
+    private final UserCourseNotificationStatusService userCourseNotificationStatusService;
+
+    private final UserRepository userRepository;
+
+    public UserCourseNotificationStatusResource(UserCourseNotificationStatusService userCourseNotificationStatusService, UserRepository userRepository) {
+        this.userCourseNotificationStatusService = userCourseNotificationStatusService;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * PUT coursenotification/courses/{courseId}/notifications/status : Update status of multiple notifications for the current user
+     *
+     * @param courseId   the ID of the course
+     * @param requestDTO the request containing the list of notification ids as well as the status
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @PutMapping("course/{courseId}/status")
```

Comment:
seems there is not security check at all here (either add a check, or in case this is intended, e.g. due to performance reasons, make sure to mention this in the JavaDOC)
also the URL does not make sense to me (sounds like the status of the course is changed) and is inconsistent with the JavaDoc above